pkg/broker/broker.go

```diff
@@ -193,26 +193,13 @@ func (a AnsibleBroker) Provision(instanceUUID uuid.UUID, req *ProvisionRequest,
 	// we're being asked to provision.
 	//
 	// if err is not nil, we will just bubble that up
-	if si, err := a.dao.GetServiceInstance(instanceUUID.String()); err == nil {
-		if si.Id.String() == serviceInstance.Id.String() &&
-			reflect.DeepEqual(si.Parameters, serviceInstance.Parameters) {
 
+	if si, err := a.dao.GetServiceInstance(instanceUUID.String()); err == nil {
+		if serviceInstance.IsEqual(si) {
 			a.log.Debug("already have this instance returning 200")
 			return &ProvisionResponse{}, ErrorAlreadyProvisioned
-		} else if si.Id.String() == serviceInstance.Id.String() &&
-			!reflect.DeepEqual(si.Parameters, serviceInstance.Parameters) {
-
-			// TODO: remove these debug statements at some point
-			a.log.Debug("Existing parameters")
-			for k, v := range map[string]interface{}(*si.Parameters) {
-				a.log.Debug("%s = %s", k, v)
-			}
-			a.log.Debug("Incoming parameters")
-			for k, v := range map[string]interface{}(*serviceInstance.Parameters) {
-				a.log.Debug(fmt.Sprintf("%s = %s", k, v))
-			}
-
-			a.log.Info("we have a duplicate instance with identical parameters, returning 409 conflict")
+		} else if si.Id.String() == serviceInstance.Id.String() {
+			a.log.Info("we have a duplicate instance with parameters that differ, returning 409 conflict")
```

Comment:
I was leaning towards just copying what we did on bind.

https://github.com/fusor/ansible-service-broker/blob/master/pkg/broker/broker.go#L351-L362

```
	if bi, err := a.dao.GetBindInstance(bindingUUID.String()); err == nil {
		if bi.Id.String() == bindingInstance.Id.String() {
			if reflect.DeepEqual(bi.Parameters, bindingInstance.Parameters) {
				a.log.Debug("already have this binding instance, returning 200")
				return &BindResponse{}, ErrorAlreadyProvisioned
			}

			// parameters are different
			a.log.Info("duplicate binding instance diff params, returning 409 conflict")
			return nil, ErrorDuplicate
		}
	}
```

This also IMO makes it easier to compare to the spec. Which lists out the conditions for returning. I.e. if the ids are the same and the parameters are the same return 200, else if the ids are the same but the parameters are different, 409.